### PR TITLE
ArborX neighbor lists

### LINKED ISSUE #40: Add NeighborTree
Use the Cabana/ArborX based neighbor list creation

### PR DESCRIPTION
**Pull request by streeve**
Add tree-based neighbor lists with Cabana-ArborX (currently experimental)

Closes #40

### CODE REVIEW

**Comment by streeve** [comment]
@dalg24 can you review? I left this out of the CI for now

File: src/inputCL.cpp [hunk 1]
```diff
@@ -180,7 +180,20 @@ void InputCL::read_args( int argc, char *argv[] )
...
+                neighbor_type = NEIGH_TREE;
             ++i;
+#ifndef CabanaMD_ENABLE_ARBORX
+            if ( neighbor_type == NEIGH_TREE )
+            {
+                if ( do_print )
+                {
+                    std::cout << "ArborX requested, but not compiled!"
+                              << std::endl;
+                }
```
Comment (dalg24) [inline src/inputCL.cpp:193]: I feel like you should print the error message unconditionally in that case.  Also you might want to put it to `std::cerr`.

File: src/neighbor_types/neighbor_tree.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef NEIGHBOR_TREE_H
+#define NEIGHBOR_TREE_H
+
+#include <neighbor.h>
+#include <system.h>
+#include <types.h>
```
Comment (dalg24) [inline src/neighbor_types/neighbor_tree.h:17]: You should prefix you headers

**Review by dalg24 — APPROVED** [review]

File: src/inputCL.cpp (hunk 1 above)
Comment (streeve) [reply src/inputCL.cpp:193]: This is only to restrict printing to one rank, but I agree on printing the error

**Comment by streeve** [comment]
Thanks, I'll add the more general improvements in a separate PR

**Pull request merged**